### ISSUE TITLE
Update Django to Version 5.2 LTS

### ISSUE THREAD

**Issue by afuetterer** [rdmorganiser/rdmo#742]
### Rationale / Begründung

After updating to Django 4.2 just recently, there is already an alpha release of Django 5.0. The official release is scheduled for 2023-12-04. We could add another test case to catch breaking changes early on. Dependabot would also send us a PR in December, but it might be nice to be prepared for this update, with Django being rdmo's main dependency.

### Affected

Admins, Devs

### Notes

In GitHub Actions:
- add django5 matrix entry, using include
- in tests step add pip install django ~5 at the end

### References / Verweise

* https://pypi.org/project/Django/#history
* https://pypi.org/project/Django/5.0a1/
* https://code.djangoproject.com/wiki/Version5.0Roadmap
* https://github.com/jefftriplett/python-github-actions-matrix-demo/blob/main/.github/workflows/actions.yml

**Comment by afuetterer** [comment]
Django 5.0 is in beta now. It only supports Python >= 3.10.
https://docs.djangoproject.com/en/dev/releases/5.0/#python-compatibility

What does this mean for rdmo? Stick with Django 4.2 to be Python 3.8 and 3.9 compatible?

**Comment by jochenklar** [comment]
Yes please, for as long as possible.

**Comment by afuetterer** [comment]
Summary: 

Django cannot be updated to 5.0, because it will drop Support for Python <= 3.10.
Django 4.2 is a LTS version, that receives extended support until 2026. Python 3.9 EOLs in 2025.

**Django 4.2 is here to stay for the next two years.**

Refs:
- https://devguide.python.org/versions/
- https://www.djangoproject.com/download/

**Comment by afuetterer** [comment]
5.2.8 is 

> Expected November 5, 2025

Ref: https://docs.djangoproject.com/en/5.2/releases/5.2.8/

PR is welcome? Or you want to stay with 4.2?

**Comment by MyPyDavid** [comment]
I would see this more as part of RDMO 3.0 (or 3.1), it is currently under development with many changes to the backend as well, probably it will take some months for the release.
However, with the end of support for 4.2 LTS in April 2026, it may be sensible to include this upgrade to Django 5.2 in the RDMO 3.0 already as well??

**Comment by jochenklar** [comment]
Yes, and we have a deadline for RDMO 3 😜.

**Comment by MyPyDavid** [comment]
yes, so that the goal is to upgrade to Django 5.2 in the RDMO 3 release and that our 'deadline' for RDMO 3 will be (hopefully) April 2026 as well 😉.
Maybe an initial PR can be useful to identify what needs to be changed and depending on that it could be merged earlier/later in the `3.0.0` release branch. But it might also become a longer lasting open PR. 
If you want start with it you can choose when! 🙏

**Comment by MyPyDavid** [comment]
there are currently many open PRs that need to be merged first, for the 2.3.3/2.4.0 it should be ready in the next few weeks. 
I guess we will start rebasing only when there are a few open PRs left.

**Comment by afuetterer** [comment]
Django 5.2.8, which adds support for Python 3.14 has been released.
Ref: https://docs.djangoproject.com/en/dev/releases/5.2.8/

**Comment by afuetterer** [comment]
Can this be closed? Django 5.2 is on main since https://github.com/rdmorganiser/rdmo/releases/tag/2.5.0.

**Comment by MyPyDavid** [comment]
yes, thanks @afuetterer , this was (already) included due to end of Django 4.2 LTS during RDMO 2.5.0
